### DISCUSSION TITLE
Collection of incompatibilities introduced in Mill 0.11 development

### DISCUSSION THREAD

**Discussion by lefou** [com-lihaoyi/mill#2462] [Development]
One comment per issue and possible solutions/migration in it's comments.

We can later create a nice documentation for this content.

**Comment by lefou** [comment]
`import $exec` no longer works. The [mill-integrationtest](https://github.com/lefou/mill-integrationtest) plugin suggests to use that construct to refer to the plugin under test.

> ```scala
> // build.sc
> import $exec.plugins

**Reply by lefou** [reply]
You can just use `import $file.plugins` instead.

**Comment by lefou** [comment]
`RunScript.resolveTasks` is gone.

**Reply by lefou** [reply]
Use `mill.resolve.Resolve.{Segments,Tasks}` instead.

**Comment by f-zl** [comment]
https://github.com/freechipsproject/chisel-template/issues/116
mill 0.11 gives the error "not found: type Tests". How should one edit build.sc to solve it?

**Reply by lefou** [reply]
The correct trait name depends on the outer trait.

In a `ScalaModule` it's `ScalaTests`, in `ScalaJSModule` it's `ScalaJSTests`, in `ScalaNativeModule` it's `ScalaNativeTests`, in `JavaModule` it's `JavaModuleTests`. If in doubt, just look for an inner trait that extends `TestModule`.

**Reply by lefou** [reply]
Mill 0.10.13 deprecates all inner `Tests` traits and introduces some type aliases.
